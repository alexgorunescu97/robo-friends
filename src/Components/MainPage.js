import React from 'react';
import CardList from '../Components/CardList';
import SearchBox from '../Components/SearchBox';
import Scroll from '../Components/Scroll';
import ErrorBoundary from '../Components/ErrorBoundary';

class MainPage extends React.Component {

    componentDidMount() {
        this.props.onRequestRobots();
    }

    filterRobots = () => {
        return this.props.robots.filter(robot => robot.name.toLowerCase().includes(this.props.searchField.toLowerCase()));
    }

    render() {
        const { searchField, onSearchChange, isPending } = this.props;
        if (isPending) {
            return (
                <h1>Loading...</h1>
            )
        }
        return (
            <div className="tc">
                <h1 className="f1">RoboFriends</h1>
                <SearchBox 
                    searchChange={onSearchChange}
                    searchfield={searchField}
                    />
                <Scroll>
                    <ErrorBoundary>
                        <CardList robots={this.filterRobots()}/>
                    </ErrorBoundary>
                </Scroll>
            </div>
        );
    }
}

export default MainPage;