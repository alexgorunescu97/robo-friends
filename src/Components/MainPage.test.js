import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false,
        error: ''
    }
    wrapper = shallow(<MainPage {...mockProps} s/>);
});

it('expect to render MainPage component', () => {
    expect(wrapper).toMatchSnapshot();
});

it('expect robots to be filtered correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 1,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'john',
        isPending: false,
        error: ''
    }
    let wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filterRobots()).toEqual([{
        id: 1,
        name: 'John',
        email: 'john@gmail.com'
    }]);
});

it('expect robots to be filtered correctly 2', () => {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 1,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'a',
        isPending: false,
        error: ''
    }
    let wrapper3 = shallow(<MainPage {...mockProps3} />);
    expect(wrapper3.instance().filterRobots()).toEqual([]);
});

it('expect to render MainPage component when the page is loading', () => {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: true,
        error: ''
    }
    let wrapper4 = shallow(<MainPage {...mockProps3} />);
    expect(wrapper4).toMatchSnapshot();
});