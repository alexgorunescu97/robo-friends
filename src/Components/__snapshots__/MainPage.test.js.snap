// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage component 1`] = `
<div
  className="tc"
>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <SearchBox
    searchfield=""
  />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;

exports[`expect to render MainPage component when the page is loading 1`] = `
<h1>
  Loading...
</h1>
`;
